pages/api/users/[username]/index.js

```diff
@@ -107,79 +107,90 @@ export async function getUserApi(req, res, username) {
 
   getProfile.events = dateEvents;
 
+  let updates = [];
   const date = new Date();
   date.setHours(1, 0, 0, 0);
 
   if (!isOwner) {
-    try {
-      await Stats.updateOne(
-        {
-          date,
-        },
-        {
-          $inc: { users: 1 },
-        },
-        { upsert: true }
-      );
-      log.info(`app profile stats incremented for username: ${username}`);
-    } catch (e) {
-      log.error(e, `app profile stats failed for ${username}`);
-    }
-
-    try {
-      await Profile.updateOne(
-        {
-          username,
-        },
-        {
-          $inc: { views: 1 },
-        }
-      );
-      log.info(`stats incremented for username: ${username}`);
-    } catch (e) {
-      log.error(
-        e,
-        `failed to increment profile stats for username: ${username}`
-      );
-    }
+    updates.push((async () => {
```

Comment:
I think this could be done without the function and await because you have the promise all?

Reply:
You need to create an async function to return a promise. Then add that promise to the array. Promise.allSettled collects all the promises in the array into a single promise that will resolve/reject once all promises are settled. Then you await that single promise.

Reply:
ok thank you 👍